### PR TITLE
GameBoard

### LINKED ISSUE #12: VariableBuilding class
Represents building that allows spending fixed number of resources of given number of resource type.

Unit tests: Yes. Be careful

### LINKED ISSUE #13: ArbitraryBuilding class
Represents building that allows 1-7 resources of any kind to be spent

Unit tests: Yes.

### LINKED ISSUE #26: ToolMakerHutsFields class
Implements these three actions and interaction among them (when the game has less players)

Unit tests: Yes. Focus on the interaction.

### LINKED ISSUE #27: PlaceOnFieldsAdaptor class
Adaptor between InterfaceFigureLocationInternal and ToolMakerHutFields for fields action.

Unit tests: No. It is an adaptor/facade class with very simple to no responsibilities. Testing is best handled by an integration test

### LINKED ISSUE #28: PlaceOnHutAdaptor class
Adaptor between InterfaceFigureLocationInternal and ToolMakerHutFields for hut action.

Unit tests: No. It is an adaptor/facade class with very simple to no responsibilities. Testing is best handled by an integration test

### LINKED ISSUE #29: PlaceOnToolMakerAdaptor class
Adaptor between InterfaceFigureLocationInternal and ToolMakerHutFields for tool maker action.

Unit tests: No. It is an adaptor/facade class with very simple to no responsibilities. Testing is best handled by an integration test

### LINKED ISSUE #31: GameBoard class
Puts game board state together.

Unit tests: No. It is an adaptor/facade class with very simple to no responsibilities. Testing is best handled by an integration test

### PR DESCRIPTION
**Pull request by Duchyna1**
resolves #26 
resolves #28 
resolves #27 
resolves #29 
resolves #12 
resolves #13

### CODE REVIEW

File: src/main/java/sk/uniba/fmph/dcs/game_board/ToolMakerHutsFields.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        toolMakerFigures.clear();
+        hutFigures.clear();
+        fieldsFigures.clear();
+    }
+
+    /**
+     * @return state of figure placment
+     */
+    public String state() {
+        return "ToolMakerHutsFields:\n"
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/ToolMakerHutsFields.java:173]: To JSON

File: src/main/java/sk/uniba/fmph/dcs/game_board/VariableBuilding.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        int sum = 0;
+        for (Effect resource : resources) {
+            sum += resource.points();
+        }
+        return OptionalInt.of(sum);
+    }
+
+    @Override
+    public String state() {
+        return "Number of resources required: " + numberOfResources + ", exact number of types required: " + numberOfResourcesTypes + "\n";
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/VariableBuilding.java:35]: To JSON

Comment (Duchyna1) [reply src/main/java/sk/uniba/fmph/dcs/game_board/VariableBuilding.java:35]: Čo znamená to JSON?

File: src/main/java/sk/uniba/fmph/dcs/game_board/SimpleBuilding.java [hunk 3]
```diff
@@ -33,9 +33,8 @@ public OptionalInt build(final Collection<Effect> resources) {
         return OptionalInt.of(sum);
     }
 
-    // TODO
     @Override
     public String state() {
-        return "TODO";
+        return "Required resources: " + requiredResources.toString() + "\n";
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/SimpleBuilding.java:38]: To JSON

File: src/main/java/sk/uniba/fmph/dcs/game_board/ArbitraryBuilding.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        int sum = 0;
+        for (Effect resource : resources) {
+            sum += resource.points();
+        }
+        return OptionalInt.of(sum);
+    }
+
+    @Override
+    public String state() {
+        return "Number of arbitrary resources required: " + numberOfResources + "\n";
```
Comment (Stanko2) [inline src/main/java/sk/uniba/fmph/dcs/game_board/ArbitraryBuilding.java:29]: To JSON

**Review by Stanko2 — CHANGES_REQUESTED** [review]

File: src/main/java/sk/uniba/fmph/dcs/game_board/VariableBuilding.java (hunk 2 above)
Comment (Stanko2) [reply src/main/java/sk/uniba/fmph/dcs/game_board/VariableBuilding.java:35]: Pozri `GamePhaseController.java`

**Review by Stanko2 — APPROVED** [review]

**Comment by Duchyna1** [comment]
@Stanko2 @johankogoodvibes môžem mergovat?

**Comment by johankogoodvibes** [comment]
Asi hej

**Pull request merged**